packages/gatsby-theme-newrelic/gatsby/wrap-page-element.js

```diff
@@ -22,12 +23,34 @@ const wrapPageElement = ({ element, props }, themeOptions) => {
 
   return (
     <I18nextProvider i18n={i18n}>
-      <GlobalStyles />
-      {element}
+      <LocaleProvider i18n={i18n}>
+        <GlobalStyles />
+        {element}
+      </LocaleProvider>
     </I18nextProvider>
   );
 };
 
+const getLocale = (props, themeOptions) => {
+  const { pageContext, location } = props;
+  const { i18n = {} } = themeOptions;
+
+  // For some reason we enter an infinite loop if we try and create localized
+  // 404 pages in gatsby-node.js. This is a hack that will use the current page
+  // location to find the locale to ensure the 404 page is translated
+  // accordingly.
+  if (!pageContext.fileRelativePath.match(/404/)) {
+    return pageContext.locale || defaultLocale.locale;
+  }
+
+  const [, base] = location.pathname.split('/');
```

Comment:
Woah, didn't know about `[, thing]` being valid...I've been doing `[_, thing]` like a fool!

Reply:
Haha its a great trick! The nice thing about `[,]` is that you don't need to waste a variable name!